lang/java/avro/src/main/java/org/apache/avro/SchemaNormalization.java

```diff
@@ -17,31 +17,60 @@
  */
 package org.apache.avro;
 
+import org.apache.avro.util.internal.JacksonUtils;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedHashSet;
 import java.util.Map;
 import java.util.HashMap;
 import java.io.IOException;
 import java.io.UnsupportedEncodingException;
 import java.security.MessageDigest;
 import java.security.NoSuchAlgorithmException;
 
-/** Collection of static methods for generating the canonical form of
- * schemas (see {@link #toParsingForm}) -- and fingerprints of canonical
- * forms ({@link #fingerprint}).
+/** Collection of static methods for generating the parser canonical form of
+ * schemas (see {@link #toParsingForm}), standard canonical form of schemas
+ * (see {@link #toCanonicalForm(Schema)} or {@link #toCanonicalForm(Schema, LinkedHashSet)})
+ * with user defined Logical Types (see {@link #setLogicalTypes(LogicalTypes...)} and
+ * fingerprints of canonical forms ({@link #fingerprint}).
  */
 public class SchemaNormalization {
 
   private SchemaNormalization() {}
 
+  private static final LinkedHashSet<String> RESERVED_PROPERTIES = new LinkedHashSet<>();
+
+  private static final LinkedHashSet<LogicalTypes> ADDITIONAL_LOGICAL_TYPES = new LinkedHashSet<>();
+
+  // Avro standard reserved properties
+  static {
+    Collections.addAll(RESERVED_PROPERTIES,
+      "name", "type", "fields", "symbols", "items", "values",
+      "logicalType", "size", "order", "doc", "aliases", "default");
+  }
+
+  // Set user defined logical types for canonical form of schema definition.
+  public static void setLogicalTypes(LogicalTypes... lts) {
+    ADDITIONAL_LOGICAL_TYPES.addAll(Arrays.asList(lts));
+  }
+
   /** Returns "Parsing Canonical Form" of a schema as defined by Avro
     * spec. */
   public static String toParsingForm(Schema s) {
-    try {
-      Map<String,String> env = new HashMap<>();
-      return build(env, s, new StringBuilder()).toString();
-    } catch (IOException e) {
-      // Shouldn't happen, b/c StringBuilder can't throw IOException
-      throw new RuntimeException(e);
-    }
+    return toNormalizedForm(s, true);
+  }
+
+  /** Returns "Standard Canonical Form" of a schema as defined by Avro
+    * spec. */
+  public static String toCanonicalForm(Schema s) {
+    return toCanonicalForm(s, new LinkedHashSet<>());
+  }
+
+  /** Returns "Standard Canonical Form" of a schema as defined by Avro
```

Comment:
Is this in the spec yet? If not, we shouldn't claim it is.  Or else we should add it there.
Also, it would be good to more thoroughly describe how the properties parameter is used.